rclpy/rclpy/parameter_client.py

```diff
@@ -0,0 +1,273 @@
+        """
+        request = GetParameterTypes.Request()
+        request.names = names
+        future = self.get_parameter_types_client_.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def set_parameters_atomically(
+        self, parameters: Sequence[Parameter], callback: Union[Callable, None] = None
+    ) -> Future:
+        """
+
+        :param parameters:
+        :type parameters: Union[Set[Parameter], Parameter]
+        :param callback:
+        :type callback: Union[Callable, None]
+        :return:
+        :rtype:
+        """
+        request = SetParametersAtomically.Request()
+        request.parameters = parameters
+        future = self.set_parameters_atomically_client_.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def delete_parameters(
+        self, names: List[str], callback: Union[Callable, None] = None
+    ) -> Future:
+        # TODO: `Static parameter cannot be undeclared`
+        """ Unset parameters with given names.
+
+        :param names:
+        :type names: List[str]
+        :param callback:
+        :type callback: Union[Callable, None]
+        :return:
+        :rtype:
+        """
+        request = SetParameters.Request()
+        request.parameters = [RclpyParameter(name=i).to_parameter_msg() for i in names]
+        future = self.set_parameter_client_.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def load_parameter_file(self, parameter_file, use_wildcard, return_parameters=False):
+        with open(parameter_file, 'r') as f:
+            param_file = yaml.safe_load(f)
+            param_keys = []
+            if use_wildcard and '/**' in param_file:
+                param_keys.append('/**')
+            if self.target_node in param_file:
+                param_keys.append(self.target_node)
+
+            if param_keys == []:
+                raise RuntimeError('Param file does not contain parameters for {}, '
+                                   ' only for nodes: {}' .format(node_name, param_file.keys()))
+            param_dict = {}
+            for k in param_keys:
+                value = param_file[k]
+                if type(value) != dict or 'ros__parameters' not in value:
+                    raise RuntimeError('Invalid structure of parameter file for node {}'
+                                       'expected same format as provided by ros2 param dump'
+                                       .format(k))
+                param_dict.update(value['ros__parameters'])
+            parameters = parse_parameter_dict(namespace='', parameter_dict=param_dict)
+            future = self.set_parameters(parameters)
+            if return_parameters:
+                return (future, parameters)
+            return future
+
+def parse_parameter_dict(namespace, parameter_dict):
```

Comment:
I would mark this function as private to this module by prepending a double underscore

```suggestion
def __parse_parameter_dict(namespace, parameter_dict):
```

Reply:
I've since moved this to `parameter.py` since it makes more sense to provide the methods for loading in parameters from a dict there as opposed to in `parameter_client`.

Reply:
Regarding private: I've prepended a single underscore since 1) it's not a class method (I don't think non-class methods get name-mangled, do they?) and 2) though I'd like it to be private it is also fairly useful if you happen to have a dict of parameters from elsewhere.